src/qweb/compilation_context.ts

```diff
@@ -156,6 +156,20 @@ export class CompilationContext {
     this.rootContext.shouldDefineScope = true;
     return compileExpr(expr, this.variables);
   }
+  captureExpression(expr: string): string {
+    const argId = this.generateID();
+    this.rootContext.shouldDefineScope = true;
+    const tokens = compileExprToArray(expr, this.variables);
+    const done = new Set();
+    return tokens.map((tok) => {
+      if (tok.varName && !done.has(tok.varName)) {
+        done.add(tok.varName);
+        this.addLine(`const ${tok.varName}_${argId} = ${tok.value};`);
+        tok.value = `${tok.varName}_${argId}`;
```

Comment:
this is not correct: the `const` should only be done once, but the `tok.value` replacement should be done for every tokens in the expression

Reply:
done + tested